reference/lib/ReferenceOrderValidator.sol

```diff
@@ -407,6 +409,9 @@ contract ReferenceOrderValidator is
 
                 // Update order status to mark the order as valid.
                 _orderStatus[orderHash].isValidated = true;
+                
+                // check i to prevents overflow.
+                unchecked {++i}
```

Comment:
There's a bug here where you only increment `i` if the order has not already been validated. So if I pass an `orders` array where `orders[i]` has been validated before, I'll get to an infinite loop and the item does not get skipped.

I also think that they have a rule of no `unchecked` and `assembly` in the reference contracts.

Reply:
Ohh yes, you are correct, will update it. Moreover, for the reference contracts rule, I didn't know about it. Thanks for letting me know.